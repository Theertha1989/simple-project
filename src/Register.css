.title {
    padding: 30px;
    color:dimgray;
}