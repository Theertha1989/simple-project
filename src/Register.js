import { Button, TextField, Container, Grid } from '@material-ui/core';
import React, { Fragment } from 'react';
import axios from 'axios';


import Radio from '@material-ui/core/Radio';
import RadioGroup from '@material-ui/core/RadioGroup';
import FormControlLabel from '@material-ui/core/FormControlLabel';
import FormControl from '@material-ui/core/FormControl';
import FormLabel from '@material-ui/core/FormLabel';

import './Register.css';

export default class Register extends React.Component {

  state = {
    username: '',
    password: '',
    dateOdBirth: '',
    gender1: '',
    age: null,
    enabled: false,
    getListEnabled: false,
    data: [],
    persons: [],
  };

  handleSubmit = (event) => {
    const { username, password, dateOdBirth, gender1, age } = this.state;
    event.preventDefault();
    this.setState({ enabled: true });
    axios.post('http://localhost:3002/register', {
      name: username,
      password: password,
      age: age,
      DOB: dateOdBirth,
      gender: gender1,
    })
    .then((response) => {
      console.log("response", response);
      const persons = response.data;
      this.setState({ persons });
    }, (error) => {
      console.log("error", error);
    });
  }

  componentDidMount() {
    axios.get(`http://localhost:3002/getCustomers`)
      .then(res => {
        const persons = res.data;
        this.setState({ persons });
      })
  }

  myChangeHandler = (event) => {
    let nam = event.target.name;
    let val = event.target.value;
    this.setState({ [nam]: val });
  }

  render() {
    return (
      <Container fixed>
        <h1 className="title">Registration Form</h1>
        <form name="register_form" onSubmit={this.handleSubmit} noValidate>
          <Grid container direction="row" justify="space-evenly" alignItems="center">
            <TextField id="standard-basic" name="username" label="Username" onChange={this.myChangeHandler} />
            <TextField id="standard-basic" name="password" label="Password" onChange={this.myChangeHandler} />
            <TextField id="standard-basic" name="age" label="Age" onChange={this.myChangeHandler} />
            <TextField
              name="dateOdBirth"
              id="date"
              label="Date Of Birth"
              type="date"
              defaultValue="2017-05-24"
              InputLabelProps={{
                shrink: true,
              }}
              onChange={this.myChangeHandler}
            />
            <FormControl component="fieldset">
              <FormLabel component="legend">Gender</FormLabel>
              <RadioGroup aria-label="gender" name="gender1" onChange={this.myChangeHandler} >
                <FormControlLabel value="female" control={<Radio />} label="Female" />
                <FormControlLabel value="male" control={<Radio />} label="Male" />
              </RadioGroup>
            </FormControl>
            <Button variant="contained" type="submit" id="submit" color="primary">Add</Button>
            <Button variant="contained" color="secondary">Cancel</Button>
          </Grid>
        </form>
        <ul>
        { this.state.persons.map(data =>
          <li id={data.id}>{data.name} {data.gender}  {data.DOB}  {data.age}</li> 
        )}
        </ul>
      </Container>
    )
  }
}
