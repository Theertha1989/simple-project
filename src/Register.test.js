import { render, screen } from '@testing-library/react';
import React from 'react';
import { shallow } from "enzyme";
import * as axios from 'axios';
import MockAdapter from 'axios-mock-adapter';
import Register from "./Register";

describe("Register", () => {

  const GET_URL = 'http://localhost:3002/getCustomers';
  const POST_URL = 'http://localhost:3002/register';

  const data = [{
    "id": 1,
    "name": "Prasad",
    "Age": 18,
    "DOB": "23/04/2012",
    "gender": "Male"
  },
  {
    "id": 2,
    "name": "Roopa",
    "Age": "28",
    "DOB": "22/03/1920",
    "gender": "Female"
  },];

  it("should render a Headline of register application", () => {
    const register = shallow(<Register />);
    expect(register.find('h1').length).toEqual(1)
  });

  it("should render a Username", () => {
    render(<Register />);
    const username = screen.getByText(/Username/i);
    expect(username).toBeInTheDocument();
  });

  it("should render a Password", () => {
    render(<Register />);
    const password = screen.getByText(/Password/i);
    expect(password).toBeInTheDocument();
  });

  it("should render a Date if birth", () => {
    render(<Register />);
    const dateOfBirth = screen.getByText(/Date Of Birth/i);
    expect(dateOfBirth).toBeInTheDocument();
  });

  it("should render a Gender", () => {
    render(<Register />);
    const gender = screen.getByText(/Gender/i);
    expect(gender).toBeInTheDocument();
  });

  it("should render a Add button", () => {
    render(<Register />);
    const add = screen.getByText(/Add/i);
    expect(add).toBeInTheDocument();
  });

  it("should render a Cancel button", () => {
    render(<Register />);
    const cancel = screen.getByText(/Cancel/i);
    expect(cancel).toBeInTheDocument();
  });

  it("should render Register Component", () => {
    const wrapper = shallow(<Register />);
    wrapper.setState({
      persons: data,
    });
    expect(wrapper).toMatchSnapshot();
  });

  it('should display existing added users', () => {
    const mock = new MockAdapter(axios);
    const wrapper = shallow(<Register />);
    const resp = {data: data};
    mock.onGet(GET_URL).reply(200, resp);
    // eslint-disable-next-line jest/valid-expect-in-promise
    axios.get(GET_URL).then(function (response) {
      expect(resp).toEqual(response.data);
    });
    expect(wrapper).toMatchSnapshot();
  });

  it('shouldnot display existing added users', () => {
    const mock = new MockAdapter(axios);
    const wrapper = shallow(<Register />);
    const resp = {data: data};
    mock.onGet(GET_URL).networkError();
    // eslint-disable-next-line jest/valid-expect-in-promise
    axios.get(GET_URL).then(function (response) {
      expect(resp).toEqual(response.data);
    });
    expect(wrapper).toMatchSnapshot();
  });

  it('should click on handle submit and user should get registerd', () => {
    const mock = new MockAdapter(axios);
    const wrapper = shallow(<Register />);
    const instance = wrapper.instance();
    const body = { name: 'Prasad', password: 'prassad@123', age: 18, DOB: '23/04/2012', gender: 'Male'};

    const resp = {data: data};
    mock.onPost('POST_URL', body).reply(200, resp);
    // eslint-disable-next-line jest/valid-expect-in-promise
    axios.post(POST_URL, body).then(function (response) {
      console.log(response.data);
      expect(resp).toEqual(response.data);
    });
    instance.handleSubmit({preventDefault: () => {}});
  });

  it('should click on my change handler', () => {
    const wrapper = shallow(<Register />);
    const instance = wrapper.instance();
    instance.myChangeHandler({ target: {
      name: 'username',
      value: 'abc'
    }});
  });
});