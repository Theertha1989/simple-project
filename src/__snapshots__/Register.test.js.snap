// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Register should display existing added users 1`] = `ShallowWrapper {}`;

exports[`Register should fetch users 1`] = `ShallowWrapper {}`;

exports[`Register should render Register Component 1`] = `ShallowWrapper {}`;

exports[`Register shouldnot display existing added users 1`] = `ShallowWrapper {}`;
